cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -143,7 +144,8 @@ public boolean handleException(PrintStream output, Exception e, int timesRetried
         } else if (e instanceof InvalidCommandException) {
           InvalidCommandException ex = (InvalidCommandException) e;
           output.printf("Invalid command '%s'. Enter 'help' for a list of commands\n", ex.getInput());
-        } else if (e instanceof DisconnectedException) {
+        } else if (e instanceof DisconnectedException || (e instanceof ConnectException
+                   && e.getMessage().equals("Connection refused"))) {
```

Comment:
I think we don't need to check the message for `ConnectException`. See the javadoc.